guides/source/plugins.md

```diff
@@ -3,62 +3,158 @@
 The Basics of Creating Rails Plugins
 ====================================
 
-A Rails plugin is either an extension or a modification of the core framework. Plugins provide:
-
-* A way for developers to share bleeding-edge ideas without hurting the stable code base.
-* A segmented architecture so that units of code can be fixed or updated on their own release schedule.
-* An outlet for the core developers so that they don't have to include every cool new feature under the sun.
+This guide is for developers who want to create a Rails plugin, in order to extend or modify the behavior of a Rails application.
 
 After reading this guide, you will know:
 
+* What Rails plugins are and when to use them.
 * How to create a plugin from scratch.
-* How to write and run tests for the plugin.
+* How to extend core Ruby classes.
+* How to add methods to `ApplicationRecord`.
+* How to publish your plugin to RubyGems.
 
-This guide describes how to build a test-driven plugin that will:
+--------------------------------------------------------------------------------
 
-* Extend core Ruby classes like Hash and String.
-* Add methods to `ApplicationRecord` in the tradition of the `acts_as` plugins.
-* Give you information about where to put generators in your plugin.
+What are Plugins?
+-----------------
 
-For the purpose of this guide pretend for a moment that you are an avid bird watcher.
-Your favorite bird is the Yaffle, and you want to create a plugin that allows other developers to share in the Yaffle
-goodness.
 
---------------------------------------------------------------------------------
+A Rails plugin is a packaged extension that adds functionality to a Rails
+application. While all engines are plugins, not all plugins are engines. Plugins
+can be lightweight — such as monkey patches, custom rake tasks, or shared
+concerns — or more complex, like a full Rails engine. The main difference lies
+in scope: plugins are typically used for smaller customizations or shared
+behavior across apps, whereas engines provide more fully-featured components
+with their own routes, models, and views.
+
+At a technical level, a plugin is a Ruby gem that’s designed to work inside a
+Rails application. It often uses a `Railtie` or `Engine` to hook into the Rails
```

Comment:
I feel that we should explain a bit what these are.

e.g. "a railtie is generally used as a simple plugin which exposes only ruby code and no views and controllers" and "an engine is usually used when you need to expose a mini rails app with controllers and views"